packages/flutter/lib/src/painting/image_decoder.dart

```diff
@@ -24,12 +24,15 @@ import 'binding.dart';
 /// in tests.
 Future<ui.Image> decodeImageFromList(Uint8List bytes) async {
   final ui.ImmutableBuffer buffer = await ui.ImmutableBuffer.fromUint8List(bytes);
-  final ui.Codec codec = await PaintingBinding.instance.instantiateImageCodecWithSize(buffer);
-  final ui.FrameInfo frameInfo;
   try {
-    frameInfo = await codec.getNextFrame();
+    final ui.Codec codec = await PaintingBinding.instance.instantiateImageCodecWithSize(buffer);
```

Comment:
If `instantiateImageCodecWithSize` ends up restoring the dispose, then there is no reason to dispose here.